File: kotlin/Problem151_ReverseWordsInString.kt
// https://leetcode.com/problems/reverse-words-in-a-string

class Solution {
    fun reverseWords(s: String): String {
        val text = s.trim().split(Regex("\\s+"))

        val reversed = text.reversed()

        return reversed.joinToString(" ")
    }
}

File: python/problem_151_reverse_words_in_string.py
# https://leetcode.com/problems/reverse-words-in-a-string/

class Solution:
    def reverseWords(self, s: str) -> str:
        s = self.remove_space(s)
        list_s = s.split()
        list_s = reversed(list_s)
        return ' '.join(list_s)

    def remove_space(self, s: str) -> str:
        return s.strip()
        
